### PR TITLE
Add stack tag support for nodejs automation api sdk

### PR DESCRIPTION
**Pull request by mrod-io**
# Description

Adds stack tag management support to the **nodejs** automation api sdk (https://github.com/pulumi/pulumi/issues/5681).

Before I move on to other language/runtime SDKs, I'd like to get a quick sanity check on this nodejs implementation, since they will follow a similar pattern 👍 . There is no changelog for now; this won't be merged as-is. Thank you! 🙏😄 

Fixes: https://github.com/pulumi/pulumi/issues/5681

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

- ~~[ ] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change~~

- [ ] ~~Yes, there are changes in this PR that warrants bumping the Pulumi Service API version~~

### CODE REVIEW

File: sdk/nodejs/automation/tag.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * TagMap is a key-value map of tag metadata associated with a stack.
+ */
+export type TagMap = { [key: string]: string };
```
Comment (mrod-io) [inline sdk/nodejs/automation/tag.ts:18]: Not strictly necessary. Workspace/LocalWorksapce/Stack methods could just return `{[key:string]: string}`. Included for client readability.

File: sdk/nodejs/automation/localWorkspace.ts [hunk 2]
```diff
@@ -499,6 +500,45 @@ export class LocalWorkspace implements Workspace {
         await this.runPulumiCmd(["config", "refresh", "--force", "--stack", stackName]);
         return this.getAllConfig(stackName);
     }
+    /**
```
Comment (mrod-io) [inline sdk/nodejs/automation/localWorkspace.ts:503]: Followed `config` methods fairly closely; they seem analagous.

File: sdk/nodejs/tests/automation/localWorkspace.spec.ts [hunk 3]
```diff
@@ -89,6 +88,45 @@ describe("LocalWorkspace", () => {
         await Stack.createOrSelect(stackName, ws);
         await ws.removeStack(stackName);
     });
+    describe("Tag methods: get/set/remove/list", () => {
```
Comment (mrod-io) [inline sdk/nodejs/tests/automation/localWorkspace.spec.ts:91]: Open to naming here. This seems to cover it, though - it's how I tested things locally. 😅

**Comment by mrod-io** [comment]
Seems I can't add reviewers. cc @RobbieMcKinstry @Frassle

**Comment by Frassle** [comment]
This looks reasonable to me.
You'll want to run `make changelog` to add a changelog entry for it.
cc @justinvp | @RobbieMcKinstry  for some extra nodejs eyes.

**Review by RobbieMcKinstry — COMMENTED** [review]
This looks great to me! Thanks @mrod-io! I'll take a closer look later but right now I have no notes.

**Comment by RobbieMcKinstry** [comment]
For context for my collegues, I'll echo what we discussed in community Slack:

We try to keep the AutoAPI library in sync across languages, so PRs that add language support in 1 language unfortunately need to add them in all languages before we can merge. @mrod-io has graciously offered to implement https://github.com/pulumi/pulumi/issues/5681 for Node, Go, and Python. I said we can add Dotnet.

**Comment by mrod-io** [comment]
Thanks for taking a look! Will open PR when with the Node, Go and Python implementations! 👍

**Comment by RobbieMcKinstry** [comment]
Reopening this PR per discussion on Community Slack. :D

**Review by RobbieMcKinstry — APPROVED** [review]

**Comment by RobbieMcKinstry** [comment]
All that's left is a CHANGELOG entry. I'll check to see if I have permission to add it to your fork. Otherwise, I'll have to ask you to add it :)

**Comment by RobbieMcKinstry** [comment]
bors retry p=1

**Pull request merged**